src/block_cache/block_mgr.c

```diff
@@ -600,8 +600,11 @@ __bm_switch_object(WT_BM *bm, WT_SESSION_IMPL *session, uint32_t objectid)
 
     current = bm->block;
 
-    /* There should not be a checkpoint in progress. */
-    WT_ASSERT(session, !S2C(session)->txn_global.checkpoint_running);
+    /*
+     * There should not be a checkpoint in progress if we are switching objects. It is allowed if we
+     * are creating the first tiered object.
```

Comment:
Is this the only time where we might switch objects outside of the checkpoint lock? It looks like we could also do it (with objectid > 1) when importing a table:

https://github.com/wiredtiger/wiredtiger/blob/8be0aa4ba3878c34dbcd956e12ecc44cba094f30/src/tiered/tiered_handle.c#L765.

Reply:
`wt_tiered_switch` is not on the call path of `wt_btree_switch_object`. It is only called via `tiered_open` from `wt_tiered_open`.

Reply:
Ah.  My mistake.  I've been staring at test triage too much and my brain is a bit fried...